src/tests/ftest/util/nvme_utils.py

```diff
@@ -0,0 +1,425 @@
+
+        # Define the arguments for the ior_runner_thread method
+        ior_cmd = IorCommand()
+        ior_cmd.get_params(self)
+        ior_cmd.set_daos_params(self.server_group, self.pool)
+        ior_cmd.daos_oclass.update(self.ior_daos_oclass)
+        ior_cmd.api.update(self.ior_apis)
+        ior_cmd.transfer_size.update(self.ior_transfer_size)
+
+        #If IOR write calculate the block size based on server % to fill up
+        if 'Write' in operation:
+            block_size = self.calculate_ior_block_size()
+            ior_cmd.block_size.update('{}'.format(block_size))
+            ior_cmd.flags.update(self.ior_flags)
+            #Store the container UUID and block size used for
+            #reading and verification purpose in future.
+            self.container_info["{}{}{}"
+                                .format(self.ior_daos_oclass,
+                                        self.ior_apis,
+                                        self.ior_transfer_size)] = [str(
+                                            uuid.uuid4()), block_size]
+        elif 'Read' in operation:
+            ior_cmd.flags.update(self.ior_read_flags)
+            #Retrieve the container UUID and block size for reading purpose
+            ior_cmd.block_size.update('{}'
+                                      .format(self.container_info
+                                              ["{}{}{}".format(
+                                                  self.ior_daos_oclass,
+                                                  self.ior_apis,
+                                                  self.ior_transfer_size)][1]))
+
+        # Define the job manager for the IOR command
+        manager = Mpirun(ior_cmd, mpitype="mpich")
+        manager.job.daos_cont.update(self.container_info
+                                     ["{}{}{}"
+                                      .format(self.ior_daos_oclass,
+                                              self.ior_apis,
+                                              self.ior_transfer_size)][0])
+        env = ior_cmd.get_default_env(str(manager))
+        manager.assign_hosts(self.hostlist_clients, self.workdir, None)
+        manager.assign_processes(self.processes)
+        manager.assign_environment(env, True)
+
+        # run IOR Command
+        try:
+            result = manager.run()
+            # Exception does not work so better to verify the proper
+            # completion too.
+            if 'Finished' not in result.stdout:
+                results.put("FAIL")
+        except CommandFailure as _error:
+            results.put("FAIL")
+
+    def calculate_ior_block_size(self):
+        """
+        Calculate IOR Block size to fill up the Server
+
+        Args:
+            None
+
+        Returns:
+            block_size(int): IOR Block size
+        """
+        #Check the replica for IOR object to calculate the correct block size.
+        _replica = re.findall(r'_(.+?)G', self.ior_daos_oclass)
```

Comment:
Can this be Tera bytes (or just in KB, bytes)?  
eg:  _replica = re.findall(r'_(.+?)T', self.ior_daos_oclass)
Looks like we are restricting users of the script.

Reply:
This is related to the Group "G" in "RP_2G1" and not related to size. Basically this will gives the block sizes based on replica used. if user want to fill 50% of 100GB with SX object it will be 50G block size. For replica 2G1, it will give 25G as we have Two replica to fill 50%.